### ISSUE TITLE
External select: list_name not in choices sheet

### ISSUE THREAD

**Issue by nicsi76** [Esri/Survey123Community#315]
I have a long list of administrative units that I would prefer to store in an external_choices spreadsheet instead of the choices sheet. But when I use the type select_one_external it seems like it's still looking for the data in the choices sheet. 

I have added the spreadsheet "external_choices" after the "choices" spreadsheet and put all my data there with the header line 
![image](https://cloud.githubusercontent.com/assets/11991954/13044560/7781ed86-d3d7-11e5-8205-439472aceb60.png)
Then the list of villages comes with all levels of administrative units, then the list of cells (without a blank row in between), then the list of sectors, etc. Since the number of levels of depending administrative units decrease with the level of the list, for example the columns cells, district, sector remain empty  for the provinces list. 

I looked at the example for external choices http://survey123.maps.arcgis.com/sharing/rest/content/items/68b8e40bdc594d0b800c8fe6e46863b5/data but can't find my mistake. Or is almost 17500 entries just too much?

Thank you for your ideas?

**Comment by nicsi76** [comment]
Additional info: It seems like the itemsets.csv file is not created. Also when I put e.g. only the districts (30 list items) the external choices are not loaded. I also tried to create the itemsets.csv file by hand but I still get the same error.

**Comment by Marie-Lou** [comment]
17,500 entries in external choices is not too much - I have 41,000 entries in external choices, and it works just fine.....

**Comment by nicsi76** [comment]
so it should work :-( Can you please tell me how you did it?

**Comment by Marie-Lou** [comment]
I've added a doc with a screenshot of the cascading selection; as well as the relevant columns in the survey, choices and external_choices tabs. In the external_choices tab I only show one StreetName, and then some of the HouseNumbers in that StreetName - obviously the total list has heaps more rows... :smiley:  If this isn't clear enough, you may need to show me how your XLS form look.
[Cascading Address Select with External Choices.docx](https://github.com/Esri/Survey123Community/files/131538/Cascading.Address.Select.with.External.Choices.docx)

**Comment by nicsi76** [comment]
That's really strange...

I tested now only with two questions, one list in choices, the other one in external choices. If I use the cascading list depending on the province chosen I don't get an error message but also my districts don't show. If I remove the constraint I get the error message `list name not in choices sheet: district`.

I don't get what I'm doing wrong, maybe you are more lucky. Thanks for having a look. 

[Formular 2.xlsx](https://github.com/Esri/Survey123Community/files/132276/Formular.2.xlsx)

**Comment by Marie-Lou** [comment]
Try this one.  I first created a cascading select just on names, to get the test the logic.  The second group is on the ID's (see the word doc to see how the survey look).  For some reason I can't get the district id to drop the decimal point, but I didn't try very hard to get rid of it  :smile:   This will give you the idea, thought - you should be able to refine it.  Let me know if you work out how to drop the decimal point!
Hope it helps!

[Province District Test.xlsx](https://github.com/Esri/Survey123Community/files/133153/Province.District.Test.xlsx)
[Province and District External Select.docx](https://github.com/Esri/Survey123Community/files/133156/Province.and.District.External.Select.docx)

**Comment by nicsi76** [comment]
Thanks a lot Marie-Lou!
It seems like the external choices have problems with having numbers as names. Unfortunately the numbers are the information needed in the database. But I did a workaround now in the way that I stored the name together with the number in the name field and labelled without the number. If my client really needs only the number in the database we can still put it there by retrieving it from the stored value. Thanks again for your help!
